contracts/contracts/DC.sol

```diff
@@ -0,0 +1,376 @@
+
+    /**
+     * @dev Emitted if setting the duration e.g. if the owner tried to se the duration less than the Minimum duration of 365 days
+     * @param duration The duration of the rental period
+     * @param reason The reason for the error
+     */
+    error setDurationFailed(uint256 duration, string reason);
+
+    constructor(InitConfiguration memory _initConfig) {
+        setBaseRentalPrice(_initConfig.baseRentalPrice);
+        setRevenueAccount(_initConfig.revenueAccount);
+        setRevenueAccount(_initConfig.revenueAccount);
+        setRegistrarController(_initConfig.registrarController);
+        setDuration(_initConfig.duration);
+        setResolver(_initConfig.resolver);
+        setReverseRecord(_initConfig.reverseRecord);
+        setFuses(_initConfig.fuses);
+        setWrapperExpiry(_initConfig.wrapperExpiry);
+    }
+
+    // admin functions
+    function setBaseRentalPrice(uint256 _baseRentalPrice) public onlyOwner {
+        baseRentalPrice = _baseRentalPrice;
+    }
+
+    function setRevenueAccount(address _revenueAccount) public onlyOwner {
+        emit RevenueAccountChanged(revenueAccount, _revenueAccount);
+        revenueAccount = _revenueAccount;
+    }
+
+    function setRegistrarController(address _registrarController)
+        public
+        onlyOwner
+    {
+        registrarController = _registrarController;
+    }
+
+    function setDuration(uint256 _duration) public onlyOwner {
+        if (_duration < MIN_DURATION) {
+            revert setDurationFailed(
+                _duration,
+                "Duration less than the minimum of 365 days"
+            );
+        }
+        duration = _duration;
+    }
+
+    function setResolver(address _resolver) public onlyOwner {
+        resolver = _resolver;
+    }
+
+    function setReverseRecord(bool _reverseRecord) public onlyOwner {
+        reverseRecord = _reverseRecord;
+    }
+
+    function setFuses(uint32 _fuses) public onlyOwner {
+        fuses = _fuses;
+    }
+
+    function setWrapperExpiry(uint64 _wrapperExpiry) public onlyOwner {
+        wrapperExpiry = _wrapperExpiry;
+    }
+
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    /**
+     * @dev `available` calls RegistrarController to check if a name is available
+     * @param name The name to be checked being registered
+     */
+    function available(string memory name) public returns (bool) {
+        return registrarControllerContract.available(name);
+    }
+
+    /**
+     * @dev `makeCommitment` calls RegistrarController makeCommitment with prepopulated values
+     * commitment is just a keccak256 hash
+     * @param name The name being registered
+     * @param owner The address of the owner of the name being registered
+     * @param secret A secret passed by the client e.g. const [secret] = useState(Math.random().toString(26).slice(2))
+     */
+    function makeCommitment(
```

Comment:
Being `view` instead of `pure` there is still a risk of being front-run by RPC node. But sometimes view-only functions are executed locally. We can leave this issue for later